File: rentalManager/Model/RentalItem.cpp
#include <Model.h>
#include <algorithm>




RentalItem::RentalItem(string itemName, double itemCost, int inStock) {
    itemName = itemName;
    itemCost = itemCost;
    inStock = inStock;
}

 void RentalItem::itemRented() {
    inStock -= 1;
}

 void RentalItem::itemReturned() {
    inStock += 1;
}



void RentalItem::addCustomer(Customer newCustomer) {
    customerList.push_back(newCustomer);
}

void RentalItem::removeCustomer(Customer toRemove) {
    auto it = find(customerList.begin(), customerList.end(), toRemove);
    if(it != customerList.end())
        customerList.erase(it, customerList.end());
}

vector<Customer> RentalItem::getCustomers() {
    return customerList;
}

int RentalItem::getItemsInStock() {
    return inStock;
}

double RentalItem::getItemCost() {
    return itemCost;
}

string RentalItem::getItemName() {
    return itemName;
}

void RentalItem::setItemCost(double newCost) {
    itemCost = newCost;
}

void RentalItem::setItemsInStock(int newStock) {
    inStock = newStock;
}


 /*void RentalItem::createRentalAgreement(Customer withCustomer, RentalItem rentalItem) {
    withCustomer.addRentalItem(rentalItem);
    rentalItem.addCustomer(withCustomer);
    rentalItem.itemRented();
}*/

File: rentalManager/Model/RentalItem.cs
using System;
using System.Collections.Generic;
using System.Linq;

class RentalItem {

        string itemName;
        double itemCost;
        int inStock;
        List<Customer> customerList = new List<Customer>();  

public RentalItem(string itemName, double itemCost, int inStock) {
    this.itemName = itemName;
    this.itemCost = itemCost;
    this.inStock = inStock;
}

 public void itemRented() {
    inStock -= 1;
}

 public void itemReturned() {
    this.inStock = this.inStock + 1;
}

public void addCustomer(Customer newCustomer) {
    customerList.Add(newCustomer);
}

public void removeCustomer(Customer toRemove) {
    var customerToRemove = customerList.SingleOrDefault(x => x.getCustomerID() == toRemove.getCustomerID());
    if (customerToRemove != null) 
        customerList.Remove(customerToRemove);
}

public List<Customer> getCustomers() {
    return this.customerList;
}

public int getItemsInStock() {
    return this.inStock;
}

public double getItemCost() {
    return this.itemCost;
}

public string getItemName() {
    return this.itemName;
}

public void setItemCost(double newCost) {
    this.itemCost = newCost;
}

public void setItemsInStock(int newStock) {
    this.inStock = newStock;
}

public void createRentalAgreement(Customer withCustomer, RentalItem rentalItem) {
    RentedItem rentedItem = new RentedItem(DateTime.Now, rentalItem);
    withCustomer.addRentalItem(rentedItem);
    rentalItem.addCustomer(withCustomer);
    rentalItem.itemRented();
}

}

 